### PR TITLE
[Feature-WIP](inverted index) inverted index reader's implementation, and add mysql_fulltext regression case to test fulltext query

### PR DESCRIPTION
**Pull request by Tanya-W**
# Proposed changes

Issue Number: Step2 of [DSIP-023: Add inverted index for full text search](https://cwiki.apache.org/confluence/display/DORIS/DSIP-023%3A+Add+inverted+index+for+full+text+search?src=contextnavpagetreemode)
implementation of inverted index reader

dependency pr: https://github.com/apache/doris/pull/14211 https://github.com/apache/doris/pull/15807 https://github.com/apache/doris/pull/15821

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/scan/vscan_node.h [hunk 1]
```diff
@@ -334,6 +334,11 @@ class VScanNode : public ExecNode {
                                                    SlotDescriptor* slot, ColumnValueRange<T>& range,
                                                    PushDownType* pdt);
 
+    template <PrimitiveType T>
+    Status _normalize_match_in_compound_predicate(vectorized::VExpr* expr, VExprContext* expr_ctx,
```
Comment (xiaokang) [inline be/src/vec/exec/scan/vscan_node.h:338]: _normalize_match_compound_predicate ?

File: be/src/vec/exec/scan/vscan_node.cpp [hunk 2]
```diff
@@ -966,6 +966,24 @@ Status VScanNode::_normalize_compound_predicate(
                             },
                             active_range);
 
+                    _compound_value_ranges.emplace_back(active_range);
+                }
+            } else if (TExprNodeType::MATCH_PRED == child_expr->node_type()) {
+                SlotDescriptor* slot = nullptr;
+                ColumnValueRangeType* range_on_slot = nullptr;
+                if (_is_predicate_acting_on_slot(child_expr, in_predicate_checker, &slot,
```
Comment (xiaokang) [inline be/src/vec/exec/scan/vscan_node.cpp:974]: not match_predicate_checker?

File: be/src/olap/reader.cpp [hunk 3]
```diff
@@ -452,6 +452,8 @@ void TabletReader::_init_conditions_param(const ReaderParams& read_params) {
         ColumnPredicate* predicate =
                 parse_to_predicate(_tablet_schema, tmp_cond, _predicate_mem_pool.get());
         if (predicate != nullptr) {
+            auto predicate_params = predicate->predicate_params();
+            predicate_params->value = condition.condition_values[0];
```
Comment (xiaokang) [inline be/src/olap/reader.cpp:456]: add comment

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 4]
```diff
@@ -17,30 +17,288 @@
...
     RETURN_IF_ERROR(_fs->exists(index_file_path, &exists));
     return exists;
 }
 
+std::vector<std::string> FullTextIndexReader::get_analyse_result(
+        const std::wstring& field_name, const std::wstring& value,
+        InvertedIndexQueryType query_type, InvertedIndexParserType analyser_type) {
+    std::vector<std::string> analyse_result;
+    std::shared_ptr<lucene::analysis::Analyzer> analyzer;
+    if (analyser_type == InvertedIndexParserType::PARSER_STANDARD) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:55]: should keep consistent with writer analyzer

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 5]
```diff
@@ -17,30 +17,288 @@
...
+        token_stream->close();
+    }
+
+    if (query_type == InvertedIndexQueryType::MATCH_ANY_QUERY ||
+        query_type == InvertedIndexQueryType::MATCH_ALL_QUERY) {
+        std::set<std::string> unrepeated_result(analyse_result.begin(), analyse_result.end());
+        analyse_result.assign(unrepeated_result.begin(), unrepeated_result.end());
+    }
+
+    return analyse_result;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:85]: std::move

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 6]
```diff
@@ -17,30 +17,288 @@
...
+    *iterator = new InvertedIndexIterator(index_meta, this);
+    return Status::OK();
+}
+
+Status FullTextIndexReader::query(const std::string& column_name, const void* query_value,
+                                  InvertedIndexQueryType query_type,
+                                  InvertedIndexParserType analyser_type,
+                                  roaring::Roaring* bit_map) {
+    std::string search_str = reinterpret_cast<const StringValue*>(query_value)->to_string();
+    LOG(INFO) << column_name
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:99]: too many INFO logs

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp [hunk 7]
```diff
@@ -17,30 +17,288 @@
...
+}
+
+Status StringTypeInvertedIndexReader::query(const std::string& column_name, const void* query_value,
+                                            InvertedIndexQueryType query_type,
+                                            InvertedIndexParserType analyser_type,
+                                            roaring::Roaring* bit_map) {
+    const StringValue* search_query = reinterpret_cast<const StringValue*>(query_value);
+    auto act_len = strnlen(search_query->ptr, search_query->len);
+    std::string search_str(search_query->ptr, act_len);
+    LOG(INFO) << "begin to query the inverted index from clucene"
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:208]: too many logs

File: be/src/vec/exec/scan/vscan_node.h (hunk 1 above)
Comment (Tanya-W) [reply be/src/vec/exec/scan/vscan_node.h:338]: `_normalize_match_in_compound_predicate` means match_predicate in compound node tree, `in` is a preposition, not means in_predicate

File: be/src/vec/exec/scan/vscan_node.cpp (hunk 2 above)
Comment (Tanya-W) [reply be/src/vec/exec/scan/vscan_node.cpp:974]: match_checker in `_normalize_match_predicate`

File: be/src/olap/reader.cpp (hunk 3 above)
Comment (Tanya-W) [reply be/src/olap/reader.cpp:456]: done

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp (hunk 4 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:55]: here reader default analyzer is SimpleAnalyzer, writer default analyzer is also SimpleAnalyzer

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp (hunk 5 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:85]: updated

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp (hunk 6 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:99]: changed to debug log

File: be/src/olap/rowset/segment_v2/inverted_index_reader.cpp (hunk 7 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/inverted_index_reader.cpp:208]: changed to debug log

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 8]
```diff
@@ -580,6 +581,10 @@ Status SegmentIterator::_apply_index_except_leafnode_of_andnode() {
         }
 
         if (!res.ok()) {
+            if (res.code() == ErrorCode::INVERTED_INDEX_FILE_NOT_FOUND &&
```
[the remark this answers is not in the document]
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:584]: done

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.54 seconds
 load time: 510 seconds
 storage size: 17122764146 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230116032920_clickbench_pr_80715.html

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**